sdk/azcore/runtime/policy_http_trace.go

```diff
@@ -103,10 +103,17 @@ func StartSpan(ctx context.Context, name string, tracer tracing.Tracer, options
 	if !tracer.Enabled() {
 		return ctx, func(err error) {}
 	}
+	if activeSpan := ctx.Value(ctxActiveSpan{}); activeSpan != nil {
+		// per the design guidelines, if a SDK method Foo() calls SDK method Bar(),
+		// then the span for Bar() must be suppressed. however, if Bar() makes a REST
+		// call, then Bar's HTTP span must be a child of Foo's span.
+		return ctx, func(err error) {}
```

Comment:
this does not need to be addressed right away, but there is an unobvious exception:

e.g. ServiceBus processor callback creates a span for processing and then the user calls into some SDK method like
- ServiceBus.process
  - BlobClient.upload


In this case no suppression is needed for `BlobClient.upload` because user explicitly called it. Span kind can be used as a heuristic to distinguish those (i.e. only suppress `client` or `internal` spans that happen in the scope of other `client` or `internal` spans.

Reply:
/cc @richardpark-msft

Reply:
@lmolkova, just to check - do _I_ need an equivalent policy/check inside of Service Bus and Event Hubs to avoid nested spans?

Reply:
I'd expect suppression to be done in core and should never be a library concern - this is what we do in other languages. 

It should probably happen inside `tracer.Start`.

Reply:
This should now be fixed.  Only suppress the span if the in-flight span is a `client` or `internal` span.